src/olympia/scanners/tasks.py

```diff
@@ -235,29 +235,35 @@ def mark_yara_query_rule_as_completed_or_aborted(query_rule_pk):
     Mark a ScannerQueryRule as completed/aborted.
     """
     rule = ScannerQueryRule.objects.get(pk=query_rule_pk)
-    if rule.state == RUNNING:
-        rule.update(state=COMPLETED)
-    elif rule.state == ABORTING:
-        rule.update(state=ABORTED)
+    try:
+        if rule.state == RUNNING:
+            rule.change_state_to(COMPLETED)
+        elif rule.state == ABORTING:
+            rule.change_state_to(ABORTED)
+    except ImproperScannerQueryRuleStateError:
+        log.error('Not marking rule as completed or aborted for rule %s in '
+                  'mark_yara_query_rule_as_completed_or_aborted, its state is '
+                  '%s' % (rule.pk, rule.get_state_display()))
 
 
 @task
 def run_yara_query_rule(query_rule_pk):
```

Comment:
We could also add some `statsd` calls to monitor things (not in this patch obviously).

Reply:
I debated adding `statsd` calls, but because code search will be used infrequently and with very different rules, I wasn't sure there would be enough data to properly visualize stuff in Grafana.